lib/Service/OpmlService.php

```diff
@@ -75,15 +79,27 @@ public function import(string $userId, string $data): bool
         }
 
         foreach ($feeds as $feed) {
+            if ($this->feedService->existsForUser($userId, $feed['url'])) {
+                continue;
+            }
             $parent = $folderEntities[$feed['folder']] ?? null;
-            $this->feedService->create(
-                $userId,
-                $feed['url'],
-                $parent?->getId(),
-                full_text: false,
-                title: $feed['title'],
-                full_discover: false,
-            );
+            try {
+                $this->feedService->create(
+                    $userId,
+                    $feed['url'],
+                    $parent?->getId(),
+                    full_text: false,
+                    title: $feed['title'],
+                    full_discover: false,
+                );
+            } catch (\Exception $e) {
```

Comment:
Let's make this specific so we don't keep going if there's catastrophic errors.

Reply:
What would be a catastrophic error here? 
You get an exception if a feed can't be added because of not found, not parse-able or whatever problem the single feed had and continue to import all working feeds.

Reply:
Database has gone away seems like an example of one that could be unintentionally caught here.

Reply:
Changed to ServiceNotFoundException | ServiceConflictException